### PR TITLE
Fixed restart query causing prepare to fail and improve performance for backend usecases

### PR DESCRIPTION
**Pull request by khanaffan**

### CODE REVIEW

File: docs/changehistory/NextVersion.md [hunk 1]
```diff
@@ -1494,7 +1494,7 @@ Now, you can bind a set of Ids as a parameter for the `IN` clause. The Ids will
 The signature of the method has changed to:
 
 ```ts
-query(ecsql: string, params?: QueryBinder, rowFormat = QueryRowFormat.UseArrayIndexes, config?: QueryOptions): AsyncIterableIterator<any>;
+query(ecsql: string, params?: QueryBinder, rowFormat = QueryRowFormat.UseECSqlPropertyIndexes, config?: QueryOptions): AsyncIterableIterator<any>;
```
Comment (grigasp) [inline docs/changehistory/NextVersion.md:1497]: Is the part `rowFormat = QueryRowFormat.UseECSqlPropertyIndexes` correct? It looks like `rowFormat` was moved into `QueryOptions`.

Comment (pmconne) [inline docs/changehistory/NextVersion.md:1497]: No longer correct - QueryRowFormat is part of QueryOptions, no?

Comment (pmconne) [reply docs/changehistory/NextVersion.md:1497]: @grigasp saw the same - I'll adjust it.

File: docs/changehistory/NextVersion.md [hunk 2]
```diff
@@ -1494,7 +1494,7 @@ Now, you can bind a set of Ids as a parameter for the `IN` clause. The Ids will
...
 ```
 
 The `rowFormat` parameter defaults to `QueryRowFormat.Array`. That format is more efficient so its use is preferred, but it differs from the previous row format. You can upgrade existing code to use the old format with minimal changes. For example, if your existing code passes query parameters as an array, change it as follows:
```
Comment (pmconne) [inline docs/changehistory/NextVersion.md:1500]: QueryRowFormat.Array never existed.

File: docs/changehistory/NextVersion.md (hunk 1 above)
Comment (pmconne) [reply docs/changehistory/NextVersion.md]: Fixed.

File: docs/changehistory/NextVersion.md (hunk 2 above)
Comment (pmconne) [reply docs/changehistory/NextVersion.md]: Fixed.

**Review by pmconne — APPROVED** [review]

**Pull request merged**